src/deprecated.js

```diff
@@ -79,9 +85,21 @@ jQuery.isNumeric = function( obj ) {
 		!isNaN( obj - parseFloat( obj ) );
 };
 
-jQuery.trim = function( text ) {
-	return text == null ?
-		"" :
-		( text + "" ).replace( rtrim, "" );
-};
+if ( String.prototype.trim ) {
+	useDefaultTrim = "_" === " \uFEFF\xA0_\uFEFF\xA0 ".trim();
```

Comment:
We do this type of things using support tests. Please create a file `src/deprecated/support.js` modeled after e.g. `src/event/support.js` and save this support test result as `support.trim`. Then, you'll need to add `./deprecated/support` to AMD dependencies of this file and read `support.trim` in the below `if`.

Also, I expect it will be shorter if you:
* check `"".trim` instead of `String.prototype.trim`
* Incorporate this check into the definition, removing the `if`

Something like:
```js
support.trim = "".trim && "_" === " \uFEFF\xA0_\uFEFF\xA0 ".trim();
```

Reply:
Also, when you do this, you'll need to update `test/unit/support.js` with support test results for every listed browser. This should be `true` for all browsers except Android 4.0. Since results for Android 4.0-4.3 are grouped at the moment, you'll need to split that into 4.0 & 4.1-4.3.

Reply:
> Is there a framework to launch the test with all the browsers to see if it behaves properly?

You can run them - after building via `grunt` - on your local browsers by passing the comma-separated `BROWSERS` environment variable and setting `GITHUB_ACTION` to anything non-empty:
```
GITHUB_ACTION=1 BROWSERS=Chrome,FirefoxHeadless grunt karma:main
```
(the `GITHUB_ACTION` part shouldn't be necessary, I need to update our Gruntfile, but for now it is what it is)

To test on IE, you need to run Windows; then you should be able to pass `IE`, a proper Karma launcher is installed.

If you want to run it on a specific version of a browser that you don't have installed locally - especially IE - then it's not possible without a service like BrowserStack. We run such tests after merge at https://swarm.jquery.org/project/jquery.

You can also run:
```
grunt karma:chrome-debug
```
and then open the printed URL to a browser you want to test and click DEBUG.

Remember to keep the browser window focused, unless it's a headless browser like with the Firefox example above.

Reply:
> a) It fails with obscure error.

These tests check that all support tests in the current browser are:
* listed in the proper results object above and match those values
* fail in at least one browser - i.e. at least in one of the lists the value is not equal to `true`

> The CI seems to verify only Chrome and Firefox, while I have to modify expected outputs for ALL the browsers. I am not really sure what safari vs safari_9_10 is, thus I can just plug "trim": true and pray it would work. I do not see much value in that type of tests.

The CI on GitHub runs only these two browsers but our full test infra at https://swarm.jquery.org/project/jquery runs them in all supported browsers. We just haven't ported all of this to GitHub Actions as there are issues with test flakiness that are easier to deal with in TestSwarm than on GitHub Actions. Those tests run post-merge so yes, we can't verify everything easily before merging other than by running tests on a browser where we expect issues manually.

That said, I expect `trim` to be supported everywhere except for Android 4.0 without issues. We've already checked this in the past which is why the support comment only mentioned Android 4.0.

Reply:
BTW, there's a WIP PR (#4803) adding support for running browser tests on BrowserStack on GitHub Actions; without BrowserStack we can only run them on headless Chrome & headless Firefox.

Reply:
BTW, there's a WIP PR (#4803) adding support for running browser tests on BrowserStack on GitHub Actions; without BrowserStack we can only run them on headless Chrome & headless Firefox.

However, there has been no movement on the PR recently.

Reply:
Ok, now I got a clean run in CI: https://github.com/vlsi/jquery/actions/runs/2663071906